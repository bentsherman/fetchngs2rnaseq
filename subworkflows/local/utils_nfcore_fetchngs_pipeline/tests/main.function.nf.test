
nextflow_function {

    name "Test Functions"
    script "subworkflows/local/utils_nfcore_fetchngs_pipeline/main.nf"
    tag "UTILS_NFCORE_FETCHNGS_PIPELINE"

    test("Function isSraId") {

        function "isSraId"

        when {
            function {
                """
                input[0] = 'DRR000774'
                """
            }
        }

        then {
            assertAll(
                { assert function.success },
                { assert snapshot(function.result).match() }
            )
        }
    }

    test("Function sraCheckENAMetadataFields [success]") {

        function "sraCheckENAMetadataFields"

        when {
            function {
                """
                input[0] = 'run_accession,experiment_accession,library_layout,fastq_ftp,fastq_md5'
                """
            }
        }

        then {
            assertAll(
                { assert function.success },
                { assert snapshot(function.result).match() }
            )
        }
    }

    test("Function sraCheckENAMetadataFields [failure]") {

        function "sraCheckENAMetadataFields"

        when {
            function {
                """
                input[0] = 'run_accession,experiment_accession,library_layout,fastq_ftp'
                """
            }
        }

        then {
            assertAll(
                { assert !function.success }
            )
        }
    }

    test("Function sraCurateSamplesheetWarn") {

        function "sraCurateSamplesheetWarn"

        then {
            assertAll(
                { assert function.success },
                { assert snapshot(function.result).match() }
            )
        }
    }

}
