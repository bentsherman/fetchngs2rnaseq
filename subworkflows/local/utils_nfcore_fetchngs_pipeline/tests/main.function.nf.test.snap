{
    "Function sraCurateSamplesheetWarn": {
        "content": null,
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-28T11:55:41.001798"
    },
    "Function sraCheckENAMetadataFields [success]": {
        "content": null,
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-28T11:55:33.679255"
    },
    "Function isSraId": {
        "content": [
            true
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-28T11:55:29.999289"
    }
}